packages/chat-client/src/components/ChatWidget/ChatWidget.tsx

```diff
@@ -0,0 +1,135 @@
+import { useChat } from '../../contexts/ChatContext';
+import { MinimizedChat } from './MinimizedChat';
+import { ExpandedChat } from './ExpandedChat';
+import { Box } from '@mui/material';
+import { useEffect } from 'react';
+
+export const ChatWidgetComponent = () => {
+  const { minimized, setMinimized, chatConfig, isEmbedded } = useChat();
+
+  // Handle iframe-specific adjustments
+  useEffect(() => {
+    // Check if we're in an iframe
+    const isInIframe = window !== window.parent;
+
+    if (isInIframe) {
+      // Add message listener for parent window communication
+      const handleMessage = (event: MessageEvent) => {
+        // Handle commands from parent window
+        if (event.data && event.data.type === 'chat-command') {
+          console.log('Received command from parent:', event.data);
+          
+          const { action } = event.data;
+          
+          switch (action) {
+            case 'open':
+              setMinimized(false);
+              break;
+            case 'minimize':
+              setMinimized(true);
+              break;
+            default:
+              console.log('Unknown command:', action);
+          }
+        }
+      };
+
+      window.addEventListener('message', handleMessage);
+      return () => window.removeEventListener('message', handleMessage);
+    }
+  }, [setMinimized]);
+
+  // Notify parent window about size changes
+  useEffect(() => {
+    if (isEmbedded && window !== window.parent) {
+      window.parent.postMessage({
+        type: 'chat-widget-resize',
+        size: minimized ? 'minimized' : 'expanded'
+      }, '*');
+    }
+  }, [minimized, isEmbedded]);
+
+  // Position styling based on config and embed mode
+  const getPositionStyle = () => {
+    // When in embed mode, take up the full space of the iframe
+    if (isEmbedded) {
+      return {
```

Comment:
Why do we have so many way of representing style? We have CSS, random objects, and now additional styling returned on this condition. At that point that should be something like changing the class between two different modes